### PR TITLE
GH-190: improve error message for constructor type ref

### PR DESCRIPTION
**Pull request by qtran-n4**
## Task
Link: https://github.com/eclipse/n4js/issues/190

### CODE REVIEW

**Comment by qtran-n4** [comment]
@yoosiba Please review. Thanks! we have decided not to touch the subtyping of constructor type ref for now and hence I have simply improved the warning message.

**Comment by qtran-n4** [comment]
The builds are green!

File: tests/org.eclipse.n4js.bugreports.tests/bugreports-tests/GH-190_WeirdErrorWarningMessageConstructorTypeRef.n4js.xt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+class D extends C<I> { }
+
+let cconstr: constructor{C};
+let dconstr: constructor{D};
+
+//XPECT warnings --> "Neither constructor{C} is a subtype of constructor{D} nor constructor{D} is a subtype of constructor{C}. The expression will always evaluate to false." at "cconstr === dconstr"
+if (cconstr === dconstr) {
+	console.log("Hi");	
+}
```
Comment (yoosiba) [inline tests/org.eclipse.n4js.bugreports.tests/bugreports-tests/GH-190_WeirdErrorWarningMessageConstructorTypeRef.n4js.xt]: new line!

**Comment by qtran-n4** [comment]
Replaced by https://github.com/eclipse/n4js/pull/195

**Pull request closed without merging**